merlin/study/celerymanager.py

```diff
@@ -0,0 +1,230 @@
+# Produced at the Lawrence Livermore National Laboratory
+# Written by the Merlin dev team, listed in the CONTRIBUTORS file.
+# <merlin@llnl.gov>
+#
+# LLNL-CODE-797170
+# All rights reserved.
+# This file is part of Merlin, Version: 1.12.1.
+#
+# For details, see https://github.com/LLNL/merlin.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+###############################################################################
+
+import os
+import subprocess
+import time
+
+import psutil
+import redis
+
+
+class WorkerStatus:
+    running = "Running"
+    stalled = "Stalled"
+    stopped = "Stopped"
+    rebooting = "Rebooting"
+
+
+WORKER_INFO = {
+    "status": WorkerStatus.running,
+    "pid": -1,
+    "monitored": 1,
+    "num_unresponsive": 0,
+}
+
+
+class CeleryManager:
+    def __init__(self, query_frequency: int = 60, query_timeout: float = 0.5, worker_timeout: int = 180):
+        """
+        Initializer for Celery Manager
+        @param int query_frequency:     The frequency at which workers will be queried with ping commands
+        @param float query_timeout:     The timeout for the query pings that are sent to workers
+        @param int worker_timeout:      The sum total(query_frequency*tries) time before an attempt is made to restart worker.
+        """
+        self.redis_connection = self.get_worker_status_redis_connection()
+        self.query_frequency = query_frequency
+        self.query_timeout = query_timeout
+        self.worker_timeout = worker_timeout
+
+    @staticmethod
+    def get_worker_status_redis_connection():
+        """
+        Get the redis connection for info regarding the worker and manager status.
+        """
+        return CeleryManager.get_redis_connection(1)
+
+    @staticmethod
+    def get_worker_args_redis_connection():
+        """
+        Get the redis connection for info regarding the args used to generate each worker.
+        """
+        return CeleryManager.get_redis_connection(2)
```

Comment:
Would it be best to turn these into context managers? That way we could do something like:
```
with CeleryManager.get_worker_status_redis_connection() as redis_connection:
    redis_connection.hgetall("manager")
```
This way we don't have to worry about forgetting to close the connection.

Reply:
@bgunnar5 Created context manager for redis_connection as it made more sense.